tests/test_domain_class_relations.py

```diff
@@ -3943,6 +3943,140 @@ def test_ili2db3_domain_class_relations_ZG_Naturschutz_und_Erholungsinfrastruktu
         for expected_relation in expected_relations:
             assert expected_relation in relations_dicts
 
+    def test_widgets_for_domain_postgis(self):
+        # Schema Import
+        importer = iliimporter.Importer()
+        importer.tool = DbIliMode.ili2pg
+        importer.configuration = iliimporter_config(importer.tool, "ilimodels")
+        importer.configuration.ilimodels = "EnumsAndCats_V1"
+        importer.configuration.dbschema = "enumsandcats_{:%Y%m%d%H%M%S%f}".format(
+            datetime.datetime.now()
+        )
+        importer.configuration.inheritance = "smart2"
+        importer.configuration.create_basket_col = False
+        importer.stdout.connect(self.print_info)
+        importer.stderr.connect(self.print_error)
+        assert importer.run() == iliimporter.Importer.SUCCESS
+
+        generator = Generator(
+            DbIliMode.ili2pg,
+            get_pg_connection_string(),
+            importer.configuration.inheritance,
+            importer.configuration.dbschema,
+        )
+
+        available_layers = generator.layers()
+        relations, _ = generator.relations(available_layers)
+
+        legend = generator.legend(available_layers)
+
+        project = Project()
+        project.layers = available_layers
+        project.relations = relations
+        project.legend = legend
+        project.post_generate()
+
+        qgis_project = QgsProject.instance()
+        project.create(None, qgis_project)
+
+        count = 0
+        for layer in available_layers:
+            if layer.name == "cat":
+                # the catalogue should be a relation reference
+                field = layer.layer.fields().field("cataloguebreed")
+                type = field.editorWidgetSetup().type()
+                self.assertEqual(type, "RelationReference")
+
+                # the enumeration with defined display expressoin should be a relation reference
+                field = layer.layer.fields().field("enumcharacterwithdispexpression")
+                type = field.editorWidgetSetup().type()
+                self.assertEqual(type, "RelationReference")
+
+                # the enumeration should be a value relation
+                field = layer.layer.fields().field("enumcharacter")
+                type = field.editorWidgetSetup().type()
+                self.assertEqual(type, "ValueRelation")
+
+                config = field.editorWidgetSetup().config()
+                assert config["Value"] == "dispName"
+
+                if Qgis.QGIS_VERSION_INT >= 34200:
+                    assert config["OrderByField"]
+                    assert config["OrderByFieldName"] == "seq"
```

Comment:
Should we test `assert not config["OrderByValue"]` here?
Not sure what happens if both are true, though.